script/generate.go

```diff
@@ -166,6 +167,16 @@ func emoji(out io.Writer, in io.Reader) error {
 		})
 	}
 
+	// We also want regional indicator symbols (flags) to be part of the Emoji
+	// table. They are U+1F1E6..U+1F1FF.
```

Comment:
Can you add link to documentation of the specification?

Reply:
http://www.unicode.org/reports/tr51/#def_emoji_flag_sequence documents how flags are constructed using Regional Indicator code points.

Regional Indicator code points are not classified as Extended_Pictographic so they don't show up in your emoji table. But for the sake of calculating the width, they behave the same as other emojis. So the simplest solution is to add them to your emoji table. Alternatively, you could the detection of Regional Indicators to all other parts of your code. That would be overkill, in my opinion, but it's up to you.

In any case, you'll want `StringLength("🇯🇵") == 2`. That's what this is for.

You'll find them in the same file (look for "Regional Indicator"):

https://unicode.org/Public/13.0.0/ucd/emoji/emoji-data.txt